backend/plonk/bn254/solidity.go

```diff
@@ -741,8 +741,8 @@ contract PlonkVerifier {
         mstore(add(mPtr, 0x140), g2_srs_1_y_0)
         mstore(add(mPtr, 0x160), g2_srs_1_y_1)
         let l_success := staticcall(sub(gas(), 2000),8,mPtr,0x180,0x00,0x20)
-        // l_success := true
-        mstore(add(state, state_success), and(l_success,mload(add(state, state_success))))
+        let res_pairing := mload(0x00)
+        mstore(add(state, state_success), and(l_success,eq(res_pairing,0x1)))
```

Comment:
Hmmm, this doesn't seem equivalent.  If `state_success` is `false` at this point (e.g. because say the quotient check failed) then you're ignoring that information?  Seems to me like you need to `and` against `mload(add(state, state_success)` and `res_pairing`.  

Also, do you really need the `eq(...,0x1)`.  The yellow paper seems to me to say that the SNARKV precompiled contract either returns `0` or `1`.

Reply:
yep you're right, somehow I manage to make mistakes at each commits^^

Reply:
All good, that's what the process is for.